Description: Set Debian SO version numbers in shared libraries
Author: Alastair McKinstry <mckinstry@debian.org>
Last-Updated: 2019-04-23
Forwarded: no

Index: flang-20181226/runtime/libpgmath/lib/CMakeLists.txt
===================================================================
--- flang-20181226.orig/runtime/libpgmath/lib/CMakeLists.txt
+++ flang-20181226/runtime/libpgmath/lib/CMakeLists.txt
@@ -240,6 +240,7 @@ get_property(TARGET_OBJECTS GLOBAL PROPE
 add_library(${LIBPGMATH_LIBRARY_NAME} SHARED ${TARGET_OBJECTS})
 add_library(${LIBPGMATH_LIBRARY_NAME}_static STATIC ${TARGET_OBJECTS})
 set_target_properties(${LIBPGMATH_LIBRARY_NAME}_static PROPERTIES OUTPUT_NAME ${LIBPGMATH_LIBRARY_NAME})
+set_target_properties(${LIBPGMATH_LIBRARY_NAME} PROPERTIES SOVERSION "0d" SONAME "libpgmath.so.0d")
 install(TARGETS ${LIBPGMATH_LIBRARY_NAME}
   LIBRARY DESTINATION lib)
 install(TARGETS ${LIBPGMATH_LIBRARY_NAME}_static
Index: flang-20181226/runtime/ompstub/CMakeLists.txt
===================================================================
--- flang-20181226.orig/runtime/ompstub/CMakeLists.txt
+++ flang-20181226/runtime/ompstub/CMakeLists.txt
@@ -22,6 +22,7 @@ set_property(TARGET ompstub_static PROPE
 set(SHARED_LIBRARY TRUE)
 add_flang_library(ompstub_shared ${OMPSTUB_SRC})
 set_property(TARGET ompstub_shared PROPERTY OUTPUT_NAME ompstub)
+set_target_properties(ompstub_shared PROPERTIES SONAME "libompstub.so.0d" SOVERSION "0d")
 set(SHARED_LIBRARY FALSE)
 
 set_target_properties(ompstub_static ompstub_shared
Index: flang-20181226/runtime/flang/CMakeLists.txt
===================================================================
--- flang-20181226.orig/runtime/flang/CMakeLists.txt
+++ flang-20181226/runtime/flang/CMakeLists.txt
@@ -489,6 +489,7 @@ add_flang_library(flang_shared
   )
 set_property(TARGET flang_shared PROPERTY OUTPUT_NAME flang)
 target_link_libraries(flang_shared ${CMAKE_BINARY_DIR}/${CMAKE_CFG_INTDIR}/lib/libflangrti.so)
+set_target_properties(flang_shared PROPERTIES SOVERSION "0d" SONAME "libflang.so.0d")
 # Resolve symbols against libm and librt
 target_link_libraries(flang_shared m rt)
 
Index: flang-20181226/runtime/flangrti/CMakeLists.txt
===================================================================
--- flang-20181226.orig/runtime/flangrti/CMakeLists.txt
+++ flang-20181226/runtime/flangrti/CMakeLists.txt
@@ -83,12 +83,12 @@ add_flang_library(flangrti_static
   )
 set_property(TARGET flangrti_static PROPERTY OUTPUT_NAME flangrti)
 
-
 set(SHARED_LIBRARY TRUE)
 add_flang_library(flangrti_shared
   ${PGC_SRC_FILES}
   ${SHARED_SOURCES}
   )
+set_target_properties(flangrti_shared PROPERTIES SOVERSION "0d" SONAME "libflangrti.so.0d")
 
 # Resolve symbols against libm
 target_link_libraries(flangrti_shared m)
